src/registry/persistent.rs

```diff
@@ -0,0 +1,54 @@
+//! Defines a persistent memory based [Registry].
+
+use std::fs::File;
+use std::io::{self, Read, Write};
+
+use sha2::{Digest, Sha256};
+
+use super::{ContentHash, Registry};
+
+/// A persistent memory based [Registry].
+pub struct PersistentRegistry {
+    /// The path to the folder where the [Registry] is stored.
+    path: String,
+}
+
+impl Registry for PersistentRegistry {
+    type Error = io::Error;
+
+    fn read(&self, hash: ContentHash) -> Result<Option<impl Read>, Self::Error> {
+        match File::open(format!("{}/{}", self.path, hash)) {
```

Comment:
If you use a PathBuf, you can use Path.join to do this.
Also for a tiny optimisation, you can create a subfolder with the two first characters of the hash, you can look at how git does it.